### PR TITLE
Mutex/auth integration

### PR DESCRIPTION
**Pull request by nfarah86**
@jtolds do you want to review this for http auth ? I'll update the example, if you approve this.

### CODE REVIEW

**Comment by nfarah86** [comment]
ready for review

File: examples/auth/main.go [hunk 1]
```diff
@@ -4,30 +4,44 @@
...
+	"bytes"
 	"fmt"
+	"net/http"
 	"os"
 
-	"storj.io/storj/netstate/auth"
-
 	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
+	//"storj.io/storj/netstate/auth"
```
Comment (jtolds) [inline examples/auth/main.go]: probably don't need to check in commented code - that's what git is for!

File: examples/auth/main.go [hunk 2]
```diff
@@ -4,30 +4,44 @@
...
 )
 
 // example of how the auth package is working.
 // see readme in auth/ for how to run
 func main() {
 
+	// set client api credentials
+	// hardcoded as an example
```
Comment (jtolds) [inline examples/auth/main.go]: technically the below is the opposite of hardcoded. hardcoded would be more like

```
req.Header.Add("X-Api-Key", "abcd43531")
```

Comment (jtolds) [reply examples/auth/main.go]: but whatever!

File: cmd/netstate-http/main.go [hunk 3]
```diff
@@ -30,6 +31,14 @@ func initializeFlags() {
 	flag.Parse()
 }
 
+// server will pull from these creds and compare with client creds
+// will switch to macroons later
+func setEnv() {
+	viper.SetEnvPrefix("API")
+	os.Setenv("API_KEY", "abc123")
```
Comment (coyle) [inline cmd/netstate-http/main.go]: is this supposed to be doing this or is it left over from testing?

File: netstate/routes/netstate-routes.go [hunk 4]
```diff
@@ -39,13 +40,23 @@ func NewNetStateRoutes(logger *zap.Logger, db *boltdb.Client) *NetStateRoutes {
 func (n *NetStateRoutes) Put(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	n.logger.Debug("entering netstate http put")
 
+	// check xapiheader sent from client to the one set
+	xAPIHeader := r.Header.Get("X-Api-Key")
```
Comment (coyle) [inline netstate/routes/netstate-routes.go]: I'm wondering if we can do something here to eliminate the duplication.
maybe
```
func (n *NetStateRoutes) Validate(r *http.Request) error {
    xAPIHeader := r.Header.Get("X-Api-Key")
    if !auth.ValidateAPIKey(xAPIHeader) {
	http.Error(w, ErrInvalidCredentials, http.StatusUnauthorized)
	n.logger.Error(ErrInvalidCredentials)
	}
    return ErrInvalidCredentials
}
```

File: cmd/netstate-http/main.go (hunk 3 above)
Comment (nfarah86) [reply cmd/netstate-http/main.go]: it's suppose to do this, for now; After i work on the auth for the GRPC server, I'll be switching to macaroons altogether.  @jtolds can confirm.

File: netstate/routes/netstate-routes.go (hunk 4 above)
Comment (nfarah86) [reply netstate/routes/netstate-routes.go]: I can move this outside as a function, and call the function from each route.

File: cmd/netstate-http/main.go (hunk 3 above)
Comment (jtolds) [reply cmd/netstate-http/main.go]: you actually don't need the os.Setenv line. ideally, when you start your program, you'll have already run `export API_KEY=abc123` in bash, so you don't need this here

**Pull request closed without merging**